File: tcpserver.c
#include<stdio.h>
#include<stdio.h>
#include<netinet/in.h>
#include<stdlib.h>
#include<string.h>
#include<sys/socket.h>
#include<sys/types.h>
#define MAX 80


void func(int connfd){
    char buff[MAX];
    int n;
    for(;;){
        bzero(buff,MAX);
        
    }
}
int main()

{
    int sockfd,connfd,len;
    struct sockaddr_in servaddr,cli;
    sockfd=socket(AF_INET,SOCK_STREAM,0);
    if (sockfd==-1){ printf("HUIQWHUIQI");
    exit(0);}
    else
    printf("connected");
    bzero(&servaddr,sizeof(servaddr));
    servaddr.sin_family=AF_INET;
    servaddr.sin_addr.s_addr=htonl(INADDR_ANY)
    servaddr.sin_port=htons(PORT);
    
    if((bind(sockfd,(SA*)&servaddr,sizeof(servaddr)))!=0){
        printf("BIND FAILED");
        exit(0);
    }
    else{
        printf("YEYEYEY");
    }
    
    if((listen(sockfd,5))!=0){
         printf("listen FAILED");
        exit(0);
        
    }
    else{
        printf("YEYEYEY");}
        len=sizeof(cli);
        connfd=accept(sockfd,(SA*)&cli,&len);
        if(connfd<0){
            printf("Conn failed");
            exit(0);
        }
        else{
            printf("Server accept client!");
        }
        func(connfd);
        close(sockfd);
        return 0;
}

File: tcp_server.py
import socket

server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)

server_socket.bind((socket.gethostname(), 12345))

server_socket.listen(5)

print("Server socket is listening")

(connection, address) = server_socket.accept()
print(f"Connection from: {address}")

start_conversation = input("Start conversation? (y/n)")
if start_conversation.lower() == "n":
    connection.send("Bye".encode())
    connection.close()
    server_socket.close()
    exit(0)

msg = connection.recv(1024).decode()

print("Client: ", msg)

while msg != "Bye":
    msg = input("Server: ")

    if msg == "Bye":
        connection.send("Bye".encode())
        break
    connection.send(msg.encode())

    msg = connection.recv(1024).decode()
    print("Client: ", msg)
connection.close()
server_socket.close()

